collect_app/src/main/java/org/odk/collect/android/widgets/DateWidget.java

```diff
@@ -225,17 +132,114 @@ public void setFocus(Context context) {
         inputManager.hideSoftInputFromWindow(this.getWindowToken(), 0);
     }
 
-
     @Override
     public void setOnLongClickListener(OnLongClickListener l) {
-        mDatePicker.setOnLongClickListener(l);
+        mDateButton.setOnLongClickListener(l);
+        mDateTextView.setOnLongClickListener(l);
     }
 
-
     @Override
     public void cancelLongPress() {
         super.cancelLongPress();
-        mDatePicker.cancelLongPress();
+        mDateButton.cancelLongPress();
+        mDateTextView.cancelLongPress();
     }
 
+    private void createDateButton() {
+        TableLayout.LayoutParams params = new TableLayout.LayoutParams();
+        params.setMargins(7, 5, 7, 5);
+
+        mDateButton = new Button(getContext());
+        mDateButton.setId(QuestionWidget.newUniqueId());
+        mDateButton.setText(R.string.select_date);
+        mDateButton.setPadding(20, 20, 20, 20);
+        mDateButton.setTextSize(TypedValue.COMPLEX_UNIT_DIP, mAnswerFontsize);
+        mDateButton.setLayoutParams(params);
+        mDateButton.setEnabled(!mPrompt.isReadOnly());
+
+        mDateButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                mDatePickerDialog.show();
+            }
+        });
+    }
+
+    private void createDateTextView() {
+        mDateTextView = new TextView(getContext());
+        mDateTextView.setId(QuestionWidget.newUniqueId());
+        mDateTextView.setPadding(20, 20, 20, 20);
+        mDateTextView.setTextSize(TypedValue.COMPLEX_UNIT_DIP, mAnswerFontsize);
+    }
+
+    private void addViews() {
+        LinearLayout linearLayout = new LinearLayout(getContext());
+        linearLayout.setOrientation(LinearLayout.VERTICAL);
+        linearLayout.addView(mDateButton);
+        linearLayout.addView(mDateTextView);
+        addAnswerView(linearLayout);
+    }
+
+    private void setDate() {
+        mDateTextView.setText(getAnswer().getDisplayText());
```

Comment:
I've decided to use the same method like in FormHierarchyActivity. We should keep this now and as I know someone is trying to solve https://github.com/opendatakit/collect/issues/424 and it should be a common functionality.